libraries/ESP8266HTTPClient/examples/ReuseConnectionV2/ReuseConnectionV2.ino

```diff
@@ -0,0 +1,80 @@
+/**
+   reuseConnectionV2.ino
+
+    Created on: 22.11.2015
+
+   This example reuses the http connection and also restores the connection if the connection is lost
+*/
+
+
+#include <ESP8266WiFi.h>
+#include <ESP8266WiFiMulti.h>
+#include <ESP8266HTTPClient.h>
+
+#ifndef STASSID
+#define STASSID "your-ssid"
+#define STAPSK  "your-password"
+#endif
+
+ESP8266WiFiMulti WiFiMulti;
+
+HTTPClient http;
+WiFiClient client;
+
+void setup() {
+
+  Serial.begin(115200);
+  // Serial.setDebugOutput(true);
+
+  Serial.println();
+  Serial.println();
+  Serial.println("Connecting to WiFi...");
+
+  WiFi.mode(WIFI_STA);
+  WiFiMulti.addAP(STASSID, STAPSK);
+
+  // wait for WiFi connection
+  while ((WiFiMulti.run() != WL_CONNECTED)) {
+    Serial.write('.');
+    delay(500);
+  }
+  Serial.println(" connected to WiFi");
+
+  // allow reuse (if server supports it)
+  http.setReuse(true);
+
+
+  http.begin(client, "http://jigsaw.w3.org/HTTP/connection.html");
+  //http.begin(client, "jigsaw.w3.org", 80, "/HTTP/connection.html");
+}
+
+void loop() {
+  for (int i = 0; i < 10; i++) {
```

Comment:
This is a potentially long for() which could block the loop for a long time. In this very specific sketch that may work, but there is a long list of cases where it's not ok to do that, and examples should show good practices. It's a bit like a game: you don't want to do too much in one frame because the fps goes down and the game freezes for the duration.
Instead, I suggest moving the counter i to be global, then on every pass through the loop() increment, then once done  with the 10 tests just skip everything. No forever loop either.

Reply:
Done.